src/blockTracker/index.js

```diff
@@ -1,6 +1,52 @@
+const { NO_ALCHEMY_WEBSOCKETS, alchemyWeb3 } = require('../constants')
 const _ = require('lodash')
 const { fetchBlock, fetchCurrentBlockNumber } = require('../utils/gethRead')
 
+class AlchemyWebsocketBlockTracker {
+  constructor() {
+    this.blocks = {}
```

Comment:
could `constructor` take a `blocksCacheSize` argument that limits how large the `blocks` cache can grow? This could help avoid a memory leak if used in a long running process. `this.blocks` could become a FIFO queue or an LRU cache

Reply:
Done, I just used your implementation from the http blockTracker. This class currently isn't in use, so not going to put too much time into optimizing.